site/content/api/Credentials-API.md

```diff
@@ -403,3 +403,27 @@ Example:
 | *key*            | *yes*     | *string*   | The Base64 encoded bytes representing the shared (secret) key. |
 
 **NB** The example above does not contain any of the `not-before`, `not-after` and `enabled` properties, thus the credentials can be used at any time according to the rules defined in [Credential Verification]({{< relref "#credential-verification" >}}).
+
+### X.509 Certificate
+
+A credential type for storing the [RFC 2253](https://www.ietf.org/rfc/rfc2253.txt) formatted *subject DN* of a client certificate that is used to authenticate the device as part of a TLS handshake.
+
+Example:
+
+~~~json
+{
+  "device-id": "4711",
+  "type": "x509-cert",
+  "auth-id": "CN=device-1,O=ACME Corporation",
+  "secrets": [{}]
+}
+~~~
+
+| Name             | Mandatory | Type       | Description |
+| :--------------- | :-------: | :--------- | :---------- |
+| *type*           | *yes*     | *string*   | The credential type name, always `x509-cert`. |
+| *auth-id*        | *yes*     | *string*   | The subject DN of the client certificate in the format defined by [RFC 2253](https://www.ietf.org/rfc/rfc2253.txt). |
+| *subject-dn*     | *yes*     | *string*   | MUST be the same as the *auth-id* value. |
```

Comment:
Is this missing in the above example json?

Reply:
this needs to be removed, I added it originally because I didn't know if we could have an array of *empty* objects as secrets. The idea was to have at least the *subject-dn* in the secret. But it turned out that using an empty object is fine and we do not need to duplicate the information.